src/cyhy_config/cyhy_config.py

```diff
@@ -0,0 +1,183 @@
+"""Finds a CyHy configuration.
+
+This module provides a function to find a CyHy configuration file.
+
+It will search in the following locations:
+ - A location passed in as a parameter
+ - Environment variables for an SSM parameter path
+ - Environment variables for a file path
+ - The current working directory
+ - The user's home directory
+ - The system's /etc directory
+"""
+
+# Standard Python Libraries
+import logging
+import os
+from os import environ
+from pathlib import Path
+import pprint
+import tomllib
+from typing import Optional, Type, TypeVar
+
+# Third-Party Libraries
+from boto3 import client
+from botocore.exceptions import ClientError
+from pydantic import BaseModel, ValidationError
+
+CYHY_CONFIG_PATH = "CYHY_CONFIG_PATH"
+CYHY_CONFIG_SSM_PATH = "CYHY_CONFIG_SSM_PATH"
+
+# Ensure the logger is under the common CyHy namespace
+logger = logging.getLogger(f"cyhy.{__name__}")
+pp = pprint.PrettyPrinter(indent=4)
+
+# Define a TypeVar that is bound to BaseModel
+T = TypeVar("T", bound=BaseModel)
+
+
+def get_config(
+    file_path: Optional[str] = None,
+    ssm_path: Optional[str] = None,
+    model: Optional[Type[T]] = None,
+) -> T | dict:
+    """Get the CyHy configuration."""
+    # First we try to find the configuration file in SSM
+    # If we can't find it there, we look for it in a file
+    config = read_config_ssm(ssm_path, model)
+    if config:
+        return config
+
+    # We didn't find the configuration in SSM, so we look for it in a file
+    config_file_path = find_config_file(file_path)
+
+    return read_config_file(config_file_path, model)
+
+
+def find_config_file(file_path: Optional[str] = None) -> Path:
+    """Find a CyHy configuration file.
+
+    Args:
+        file_path: A path to a configuration file.
+
+    Returns:
+        A path to a CyHy configuration file.
+
+    Raises:
+        FileNotFoundError: If no configuration file is found.
+    """
+    # Check if the provided path exists
+    if file_path:
+        if Path(file_path).exists():
+            logger.debug("Using configuration file passed as parameter: %s", file_path)
+            return Path(file_path)
+        else:
+            logger.warning(
+                "Configuration file passed as parameter not found: %s", file_path
+            )
+
+    # Check environment variable for file path
+    env_file_value = environ.get(CYHY_CONFIG_PATH, None)
+    if env_file_value:
+        env_path = Path(env_file_value)
+        if env_path.exists():
+            logger.debug(
+                "Using configuration file from environment variable: %s", env_path
+            )
+            return env_path
+        else:
+            logger.warning(
+                "Configuration file from environment variable not found: %s", env_path
+            )
+
+    # Check the current working directory
+    cwd_path = Path("cyhy.toml")
```

Comment:
Should `cyhy.toml` be defined as a constant since we use it in several places here?

Reply:
See:
- [c88c22b](https://github.com/cisagov/cyhy-config/pull/5/commits/c88c22b6c5b4c8dc69b4cb8501bf0f2f0494b206)

Reply:
See:
- [927e787](https://github.com/cisagov/cyhy-config/pull/5/commits/927e78747d801bd846dfaacdb2dee043554b5cbc)